cli/src/main/java/keywhiz/cli/commands/DescribeAction.java

```diff
@@ -83,20 +84,49 @@ public DescribeAction(DescribeActionConfig describeActionConfig, KeywhizClient c
         break;
 
       case "secret":
-        SanitizedSecret sanitizedSecret;
-        try {
-          sanitizedSecret = keywhizClient.getSanitizedSecretByName(name);
-
-          printing.printSanitizedSecretWithDetails(sanitizedSecret);
-        } catch (NotFoundException e) {
-          throw new AssertionError("Secret not found.");
-        } catch (IOException e) {
-          throw Throwables.propagate(e);
-        }
+        describeSecret();
         break;
 
       default:
         throw new IllegalArgumentException("Invalid describe type specified: " + firstType);
     }
   }
+
+  private void describeSecret() {
+    SanitizedSecret sanitizedSecret = getSanitizedSecret();
+    List<SanitizedSecret> deletedSecrets = describeActionConfig.includeDeleted
+        ? getDeletedSecrets()
+        : null;
+
+    if (sanitizedSecret != null) {
+      printing.printSanitizedSecretWithDetails(sanitizedSecret);
+    }
+
+    if (deletedSecrets != null) {
+      printing.printDeletedSecretsWithDetails(deletedSecrets);
+    }
+  }
```

Comment:
We might want to push the null checks down a level so that the individual methods are more resilient/defensive and shorten this method up a bit. 

Another thought is that we may want to print out the main secret's details before we attempt to fetch the deleted secrets in case that fetch call is slow. That would allow you to move all of the deleted-related logic within an if check around the includeDeleted flag, keeping it closer together.

Reply:
you mean handling null in the print methods? that would allow us to print something when there's no result, which might be helpful